app/models/post.rb

```diff
@@ -0,0 +1,10 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  belongs_to :author, class_name: 'User', counter_cache: :post_counter
```

Comment:
- [ ] Why do you have `belongs_to :author, class_name: 'User'` which means a post belong to an author which is a User, and at the same time you have `belongs_to :user`? I suggest you remove the `belongs_to :user` to avoid confusion or error in your program. :blush: